### DISCUSSION TITLE
Daskifying viewshed

### DISCUSSION THREAD

**Discussion by yiftachn** [makepath/xarray-spatial#710] [Q&A]
Hey,
I considered taking on myself Daskifying the viewshed computation.
I saw that the functions implemented Dask followed the schema of:
_func = partial(numpy_func,..)
data.map_overlap(_func). 

Is it the whole deal im viewshed as well? 
If so, why hasnt it been done already as it seems like 5 minutes work.
Thanks!

**Comment by thuydotm** [comment]
We can use `map_overlap` with algorithms that can be run in parallel. Imagine dividing the input raster into many chunks and compute the output chunk by chunk independently. With the current cpu viewshed implementation, output value of a pixel in a chunk can be calculated based on the input and also the output of some pixels in some other chunks. To daskify viewshed with `map_overlap`, we need to rethink the algorithm to make it parallelizable.

**Reply by yiftachn** [reply]
Thanks! I'll try to look it over.
What is the algorithm current cpu_viewshed implements?

**Reply by thuydotm** [reply]
viewshed was originally ported from GRASS:
https://github.com/OSGeo/grass/blob/master/raster/r.viewshed/viewshed.cpp

You can find here some additional background on this function:
http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.76.4282&rep=rep1&type=pdf

**Comment by brendancol** [comment]
@yiftachn daskified viewshed now exists
